### DISCUSSION TITLE
New implementations

### DISCUSSION THREAD

**Discussion by Tron74** [frntc/RAD#12] [Ideas]
Hi, congratulations on the project.
I think it could be a good basis to implement 3 ideas already seen on other systems:

1) On the Atari 400 series I saw a cartridge with raspberry pi where you can access the Linux console directly from the keyboard of the retro computer.

2) On Micro BBC instead I saw a cartridge always with raspberry pi in which the ARM board could be used as a coprocesor

3) On the Amiga instead there is the PI Storm, so use the raspberry pi board to emulate a faster version of the processor and video card of the retrocomputer.

I believe that all features can only be implemented with a software configuration of the RAD os.

In particular, I also want to release some ideas:
- Access to the console directly on screen or indirectly, for example via terminal via geos;
- Use the native ARM coprocessor to interpret for example modern languages like python;
- Use the power of the ARM cpu to emulate for example the SuperCPU cartridge;
- On the c64 there is no external video card, but it would be possible to create new graphics modes (apologies to the c64 purists for this idea)

**Comment by frntc** [comment]
Hi, thanks!

I think the BBC Micro was designed to have coprocessors. You can try similar things with the RAD (or Sidekick64), but it's always a chicken and egg problem: designing a coprocessor without software, software without coprocessor? :-)

About the Linux console: well, something like that would be possible (as long as the RPi in the RAD does not run Linux -- running an OS just makes reliable communication on the bus impossible), but it's not really what I would like to do. But hey, there's source code ;-)

Faster CPU: yes, I've done that, but it's not yet included in the releases :-) The SuperCPU is a LOT of work, not sure if I have the time and motivation to do that. 

External video: would be easy, also with the Sidekick64. In principle Sidekick20 is doing something like what you suggested: it emulates a VIC-I and outputs to HDMI. New graphics modes would work similarly.
